doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,229 @@
+## `GradBoosting`
+
+The `GradBoosting` class implements a Gradient Boosting model that supports
+numerical and categorical features. The class offers several template 
+parameters and several runtime options that can be used to control the 
+model.
+
+Gradient Boosting is a very powerful ensemble algorithm used for both 
+classification and regression tasks. It utilises a series of weak learners 
+(eg. Decision Stumps) to arrive closer and closer to a targeted label. Each 
+weak learner is trained on the error of the previous learner, thereby 
+reducing the error of the overall model.
+
+#### Simple usage example:
+
+```c++
+// Train a decision tree on random numeric data and predict labels on test data:
+// All data and labels are uniform random; 10 dimensional data, 5 classes.
+// Replace with a data::Load() call or similar for a real application.
+// numModels is a hyperparameter refering to the number of weak learners
+arma::mat dataset(10, 1000, arma::fill::randu); // 1000 points.
+arma::Row<size_t> labels =
+    arma::randi<arma::Row<size_t>>(1000, arma::distr_param(0, 4));
+arma::mat testDataset(10, 500, arma::fill::randu); // 500 test points.
+size_t numModels = 10;
+
+mlpack::GradBoosting gb;               // Step 1: create model.
+gb.Train(dataset, labels, 5, numModels);          // Step 2: train model.
+arma::Row<size_t> predictions;
+tree.Classify(testDataset, predictions); // Step 3: classify points.
```

Comment:
is this example working? did you test it on your hardware?

Reply:
Technically it should be tested by the documentation build CI job! :smile:  I did not check the output yet, but that job should build all documentation in the repository to make sure the Markdown compiles, then it checks all the URLs, and then it also extract all code snippets and try to compile them too.  If any part of the build fails though, it stops, so if URLs did not resolve or had problems, then the part that extracts code snippets will not run.